corehq/apps/sms/tasks.py

```diff
@@ -162,6 +165,22 @@ def get_connection_slot_lock(phone_number, backend, max_simultaneous_connections
     return client.lock(key, timeout=60)
 
 
+def passes_trial_check(msg):
+    if msg.domain and domain_is_on_trial(msg.domain):
+        with CriticalSection(['check-sms-sent-on-trial-for-%s' % msg.domain], timeout=60):
+            key = 'sms-sent-on-trial-for-%s' % msg.domain
+            expiry = 90 * 24 * 60 * 60
+            client = get_redis_client()
+            value = client.get(key) or 0
```

Comment:
i believe the redis client pickles the values which allows storing many different kinds of types in the cache (even objects). but i know in this case it returns int if you store an int or returns a string if you store a string because it's a paradigm i've used before